homeassistant/components/google_assistant/__init__.py

```diff
@@ -67,10 +67,11 @@
         websession = async_get_clientsession(hass)
         try:
             with async_timeout.timeout(5, loop=hass.loop):
+                agent_user_id = call.context.user_id or DEFAULT_AGENT_USER_ID
                 res = await websession.post(
                     REQUEST_SYNC_BASE_URL,
                     params={'key': api_key},
-                    json={'agent_user_id': call.context.user_id})
+                    json={'agent_user_id': agent_user_id})
```

Comment:
This is also noted in the original documentation as well.  Failing to set a consistent Agent User ID causes the user to relink assistant so they can issue a sync command.

https://www.home-assistant.io/components/google_assistant/#troubleshooting-the-request_sync-service

I think hard-coding it is a much better approach so the user doesn't need to worry about it at all.

Reply:
It is fine to supply a const user id for now, since we don't have permission system in place yet. All users will get same access anyway. However, after we introduce permission, thing will become interesting.

Just a reminder, we need a way to supply `context.user` to system triggered event, not only for this particular issue.

Reply:
Rethink it again. I should not revert user id to a fixed value only because it breaks automation script. Instead, I am going to add an optional field to the service call, so if user need it, can be provided in the automation.

Reply:
What is the use case of supporting multiple agent_user_id ? It's linked to a test skill and this is the only instance?

Reply:
If you have multiple Google users, they can link to different HA user. If those HA users hav e different permission, they will get different devices linked in Google Assistant.

Reply:
good point.

In that case we should have this service fail if not called from the context of a user?